cmd/podman/common/build.go

```diff
@@ -65,11 +65,11 @@ func DefineBuildFlags(cmd *cobra.Command, buildOpts *BuildFlagsWrapper) {
 
 	// --pull flag
 	flag := budFlags.Lookup("pull")
-	if err := flag.Value.Set("true"); err != nil {
+	flag.DefValue = "missing"
+	if err := flag.Value.Set("missing"); err != nil {
```

Comment:
I think this is a breaking chnage, DefValue is used to allow just specifing the flag without value, i.e. `podman build --pull .` and you are now changing that default to missing. I don't think that is correct and what the users would expect if juts --pull is given.
I think this is important for docker compatibility as they have --pull as bool flag defined.

Reply:
```diff
diff --git a/cmd/podman/common/build.go b/cmd/podman/common/build.go
index c9fea58abac9..796604a6ca2f 100644
--- a/cmd/podman/common/build.go
+++ b/cmd/podman/common/build.go
@@ -301,6 +301,8 @@ func buildFlagsWrapperToOptions(c *cobra.Command, contextDir string, flags *Buil
        if flags.PullNever || strings.EqualFold(strings.TrimSpace(flags.Pull), "never") {
                pullPolicy = buildahDefine.PullNever
        }
+       logrus.Errorf("Pull value: %q", flags.Pull)
+       logrus.Errorf("Pull policy: %q", pullPolicy)
 
        var cleanTmpFile bool
        flags.Authfile, cleanTmpFile = buildahUtil.MirrorToTempFileIfPathIsDescriptor(flags.Authfile)
```

I think it's working correctly.  At least that's what the upper patch suggests when running the commands below:
```
podman (build-docs) $ ./bin/podman build --pull=missing /tmp    
ERRO[0000] Pull value: "missing"                                
ERRO[0000] Pull policy: "missing"                               
[...]

podman (build-docs) $ ./bin/podman build --pull=never /tmp      
ERRO[0000] Pull value: "never"                                  
ERRO[0000] Pull policy: "never"                                 
[...]                                              

podman (build-docs) $ ./bin/podman build --pull /tmp            
ERRO[0000] Pull value: "true"                                   
ERRO[0000] Pull policy: "always"                                          
[...]
```

Reply:
Looks like you are right, I confused DefValue for NoOptDefVal.
The pull policy selection looks very confusion and buggy. `--pull=false` is translated to missing the samne for all invalid options `--pull=abc` but that is of course out of scope for this PR so this change alone LGTM